client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx

```diff
@@ -20,18 +22,18 @@ class NameserversToggle extends PureComponent {
 	render() {
 		return (
 			<Card compact className="name-servers__dns">
-				<span className="name-servers__title">
-					{ this.props.translate( 'Use WordPress.com Name Servers' ) }
-				</span>
-
 				<form className="name-servers__toggle">
 					<ToggleControl
 						id="wp-nameservers"
 						name="wp-nameservers"
 						onChange={ this.handleToggle }
 						checked={ this.props.enabled }
 						value="active"
+						label={ this.props.translate( NAMESERVERS_TOGGLE_TITLE ) }
```

Comment:
This string will not end up being translated properly. The problem comes from how we manage translations, let me elaborate:

1) We use `translate()` and `__()` and other localization functions around our strings we want to translate (e.g. localize).
2) Automated jobs scan the code for new strings using those specific function calls, and consider them as strings for translation. Those are then added to the queue for translation.
3) Once those strings are translated, and we serve the translations together with this code, and display the translated code.

However, there is one important requirement for 1) - it does performs static analysis on the code, and it can't run the code, so `NAMESERVERS_TOGGLE_TITLE` will not be evaluated, and `this.props.translate( NAMESERVERS_TOGGLE_TITLE )` will end up improperly localized and not translated.

So, in a nutshell, we should always hardcode the strings inside the localization functions. Of course, those strings can have dynamic parts (components, variables, etc.) but there's a different syntax for that.

Let me know if that makes sense.

Reply:
Completely makes sense, Thank you for this write up. Frankly I did not know this. Now I know. I am beginning to see what I do not know 🙌🏽 Masterclass!!